File: Lab 5_Bundlers/src/TechBundle.java
import java.util.ArrayList;
import java.util.List;

public class TechBundle extends Bundle {

    public TechBundle() {
        super("Tech Bundle", "A collection of tech products", 100);  // $100 discount
    }

    @Override
    public List<Product> createBundle() {
        List<Product> products = new ArrayList<>();
        products.add(new SimpleProduct("Laptop", 1000, "A high-performance laptop"));
        products.add(new SimpleProduct("Phone", 600, "A latest-model smartphone"));
        return products;
    }
}

File: Lab5_Bundlers/Lab5_Bundlers/TechBundle.cs
using System;
using System.Collections.Generic;

namespace Lab5_Bundlers
{
    internal class TechBundle : Bundle
    {
        public TechBundle(string name, decimal discount) : base(name, discount)
        {
            // Build the bundle immediately upon creation
            _products = BuildBundle();
        }

        protected override List<IProduct> BuildBundle()
        {
            List<IProduct> bundle = new List<IProduct>();

            // Add various products to the bundle
            var smartphone = new Product("Smartphone", "Latest model with advanced features", 799.99M);
            var laptop = new Product("Laptop", "High-performance laptop for gaming", 1499.99M);
            var tablet = new Product("Tablet", "Portable tablet with 10-inch display", 499.99M);
            var headphones = new Product("Headphones", "Noise-cancelling wireless headphones", 199.99M);

            // Add products to the bundle
            bundle.Add(smartphone);
            bundle.Add(laptop);
            bundle.Add(tablet);
            bundle.Add(headphones);

            return bundle;
        }
    }
}
